genesyscloud/resource_genesyscloud_telephony_providers_edges_phone.go

```diff
@@ -0,0 +1,393 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mypurecloud/platform-client-sdk-go/v46/platformclientv2"
+	"log"
+)
+
+func resourcePhone() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Phone",
+
+		CreateContext: createWithPooledClient(createPhone),
+		ReadContext:   readWithPooledClient(readPhone),
+		UpdateContext: updateWithPooledClient(updatePhone),
+		DeleteContext: deleteWithPooledClient(deletePhone),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the entity.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"state": {
+				Description:  "Indicates if the resource is active, inactive, or deleted. Valid values: active, inactive, deleted.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "active",
+				ValidateFunc: validation.StringInSlice([]string{"active", "inactive", "deleted"}, false),
+			},
+			"site_id": {
+				Description: "The site ID associated to the phone.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"phone_base_settings_id": {
+				Description: "Phone Base Settings ID.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"line_base_settings_id": {
+				Description: "Line Base Settings ID.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"phone_meta_base_id": {
+				Description: "Phone Meta Base ID.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"web_rtc_user_id": {
```

Comment:
If this attribute references a user ID, it should be added to the `RefAttrs` for this resource so exports properly use a user reference rather than the raw ID.